assets/aml-benchmark/components/src/dataset_preprocessor/dataset_preprocessor.py

```diff
@@ -0,0 +1,135 @@
+ENV = jinja2.Environment()
+ENV.globals.update(zip=zip)
+
+
+class DatasetPreprocessor(object):
+    """DatasetPrerprocessor object class."""
+
+    def __init__(
+        self,
+        input_dataset: str = None,
+        template: str = None,
+        user_preprocessor: str = None,
+        encoder_config: str = None,
+        output_dataset: str = None
+    ):
+        """Dataset Preprocessor Class.
+
+        :param input_dataset: Path to the directory to load the dataset.
+        :param template: A json dictionary where key is the name of the column enclosed in " " and associated \
+            dict value is presented using jinja template logic which will be used to extract the \
+            respective value from the dataset.
+        :param user_preprocessor: Path to the custom preprocessor python script provided by user.
+        :param encoder_config: JSON serialized dictionary to perform mapping. Must contain key-value pair \
+            "column_name": "<actual_column_name>" whose value needs mapping, followed by key-value pairs containing \
+            idtolabel or labeltoid mappers. Example format: \
+            {"column_name":"label", "0":"NEUTRAL", "1":"ENTAILMENT", "2":"CONTRADICTION"}. This is not applicable to \
+            custom scripts.
+        :param output_dataset: Path to the jsonl file where the processed data will be saved.
+        """
+        self.input_dataset = input_dataset
+        self.template = template
+        self.user_preprocessor = user_preprocessor
+        self.encoder_config = encoder_config
+        self.output_dataset = output_dataset
+
+    def __post_init__(self) -> None:
+        """Post init call."""
+        self._validate()
+
+    def validate(self) -> None:
+        """Validate the parameters."""
+        if self.input_dataset is None:
+            mssg = (
+                "Path to load the dataset is not provided."
+            )
+            raise BenchmarkValidationException._with_error(
+                AzureMLError.create(BenchmarkValidationError, error_details=mssg)
+            )
+        if self.template is None and self.user_preprocessor is None:
+            mssg = (
+                "Please provide the input to apply preprocessing logic either via template input or \
+                script_path."
+            )
+            raise BenchmarkValidationException._with_error(
+                AzureMLError.create(BenchmarkValidationError, error_details=mssg)
+            )
+        if self.user_preprocessor and not self.user_preprocessor.endswith('.py'):
+            mssg = (
+                "Please provide python script containing your custom preprocessor logic."
+            )
+            raise BenchmarkValidationException._with_error(
+                AzureMLError.create(BenchmarkValidationError, error_details=mssg)
+            )
+
+    def add_json_filter(self, template) -> str:
+        """Add tojson filter in the template."""
+        patt = r'({{.*?}})'

```

Comment:
1. There are a lot of variable names like `patt`, `pat`, `s1`. Please update them to use proper variable names.
2. Please add a comment about what this regular expression `r'({{.*?}})'` is for.

Thanks!

Reply:
patt, pat and s1 are logic variables. They are not exposed to customer at all. So, I don't understand the point of explaining what each variable do. I mean if someone wants to uunderstand, they can debug it to understand what gets passed in another from former. Isn't this how we have been doing?

Reply:
Done. Added the required explanations as discussed.